v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/build-system.ts

```diff
@@ -118,11 +131,44 @@ export class SolidityBuildSystemImplementation implements SolidityBuildSystem {
 
     const compilationJobs = [...new Set(compilationJobsPerFile.values())];
 
-    // TODO: Filter the compilation jobs based on the cache
+    await Promise.all(
+      compilationJobs.map(async (compilationJob) =>
+        compilationJob.getBuildId(),
+      ),
+    );
```

Comment:
Computing build IDs is time consuming as it involves hashing all the compiler input sources. That's why we do it asynchronously. Once a compilation job build ID is computed, it will be returned straight away on the subsequent requests.